Validation of phone numbers based on country

smarshy opened the pull request:
Earlier, there was no validation for phone numbers with respect to the country. This PR attempts to do that with the help of [python-phonenumbers](https://github.com/daviddrysdale/python-phonenumbers) package. I tested this change successfully for a few phone numbers (mostly India based) and verified it with [javascript](https://rawgit.com/googlei18n/libphonenumber/master/javascript/i18n/phonenumbers/demo-compiled.html) / [java](http://libphonenumber.appspot.com/) demo. Country codes available at [geonames](http://www.geonames.org/countries/)

There are two dependencies required - django-cities-light and phonenumbers.

For cities-light installation - [here](https://github.com/yourlabs/django-cities-light)
`pip install django-cities-light`
Adding to settings (already done)
`python manage.py migrate`  - cities light supports version of django 1.7 and above
`python manage.py cities_light`- populating database, this can take time

For phonenumbers `pip install phonenumbers` should work

django-cities-light has been used to get the country code based on entered country and then later the phone number is validated for that region. I have kept the phone validation script in the registration app, as it was called during both volunteer and admin registration.

This PR assumes that validation of countries and cities is done (I already saw a PR for this) and that a valid country is selected.

Fixes #129

Please review and let me know if something is wrong. I searched and narrowed down to this package as it seems to solve the problem with proper validation. However, if there is a better alternative, please suggest changes

tapasweni-pathak commented on vms/vms/settings.py:
```diff
@@ -61,8 +62,8 @@
+        'USER': 'vmsadmin',
+        'PASSWORD': '0xdeadbeef',
         'HOST': 'localhost',
     }
```
@smarshy Please make a separate commit for this change. This is not related to the issue that you want to fix in this pull request.

smarshy replied:
@tapasweni-pathak I have made a mistake here.....shouldn't have pushed changed credentials. I will change that

tapasweni-pathak commented:
@smarshy While reviewing the pull request, I have merged some. Your pull request is not a bit outdated for example it is still having the placeholder in admin_signup as +999999999. Can you please fresh pull and rebase your changes?

smarshy commented:
@tapasweni-pathak I see that there are still many previous pull requests pending . Should I wait for some time and rebase later (after some more PRs are merged ) ?

tapasweni-pathak commented:
@smarshy No need. You can go ahead, most of them needs changes, only after they do the suggested changes, I'll review them. So you should go ahead.

smarshy commented:
@tapasweni-pathak  I have never rebased anything before so please check if it has been done right.  I have also added placeholder, regex for volunteer sign up. Please let me know if there are any other changes to be made and if I should add checks for the country too ( right now it is assuming that a valid one is entered/selected from the dropdown)

tapasweni-pathak commented on vms/vms/settings.py:
```diff
@@ -61,8 +62,8 @@
-        'USER': 'vmsadmin',
-        'PASSWORD': '0xdeadbeef',
+        'USER': 'myuser',
+        'PASSWORD': 'mypassword',
```
@smarshy Please remove this change from the PR.

smarshy commented:
@tapasweni-pathak I just interactively rebased  :smiley: and removed the commit. However, I have a feeling that everyone might run into errors at this point as it still assumes that validation of countries and cities is done and django cities is installed properly

tapasweni-pathak commented:
@smarshy Did you do any other PR for the same issue, or do I have to review this only?

smarshy commented:
@tapasweni-pathak I have to rebase and update this, plus remove that commit. I will do that today

smarshy commented:
@tapasweni-pathak I have rebased and updated this. It should validate for existing countries in the database. Please review.
For missing countries/dummy values it showed an error page. Since dropdowns for all locations (country,city,address) haven't been implemented yet and many people might be putting dummy values for testing, showing up an error page did not seem right. So I have chenged the code a bit to still allow registration for dummy values irrespective of the phone number ( Would this be okay? )

tapasweni-pathak commented:
@smarshy When I did `python manage.py cities_light`. I got [this](https://gyazo.com/b221816de0775ae487b8557e11d9c26e). Is there something I'm doing wrong?

smarshy commented:
@tapasweni-pathak No, it seems right. I am really confused as to why this is happening. The steps are same as documented here - [http://django-cities-light.readthedocs.org/en/stable-3.x.x/](http://django-cities-light.readthedocs.org/en/stable-3.x.x/). Is the Django version 1.7 and above?

tapasweni-pathak commented:
@smarshy I didn't add cities_light in INSTALLED_APPS in settings.py file. That was required.

smarshy commented:
@tapasweni-pathak oh, I took that for granted since I had pushed the updated settings.py....I assumed the entire branch would be pulled while testing the PR. Sorry for missing that detail.

tapasweni-pathak commented:
@smarshy It will be. It is the case that one need to add it even before testing the code. No issues. Take it as learning, it will help while writing the steps in future. Make the steps as easy as possible for others by including as much details as possible, obviously the relevant one. :)

Did you get [this](https://gyazo.com/e568168b613dddc402afb815f3cf422f)? Look at the statement, *No handles could be found....*

smarshy commented:
@tapasweni-pathak Hi, yes I did run into this. It is fairly common with django-cities-light. Here - [https://github.com/yourlabs/django-cities-light/issues/93](https://github.com/yourlabs/django-cities-light/issues/93). The database still gets populated with country, cities, region tables. There won't be any logging though. If logs are required for cities light, then logging has to be configured separately. Are logs needed?

tapasweni-pathak commented:
@smarshy Cool. At this moment, no logs are not needed.

tapasweni-pathak commented:
> This PR assumes that validation of countries and cities is done (I already saw a PR for this) and that a valid country is selected.

Till the issue is open (#75), consider it is not implemented in VMS. So we are not doing it till now in VMS.

smarshy commented:
@tapasweni-pathak Thanks :) Sure, would love to help out with testing and maintaining these changes in future. Since these involved installing software, should I propose a few changes in the docs so that others don't run into errors after pulling this? -  The version also needs an update to 1.7 for cities-light to work
